ably.ts.d

```diff
@@ -0,0 +1,457 @@
+    name: String;
+    timestamp: number; // Time
+  }
+
+  export declare class PresenceMessage {
+    fromEncoded: (JsonObject: any, channelOptions?: ChannelOptions) => PresenceMessage;
+
+    fromEncodedArray: (JsonArray: Array<any>, channelOptions?: ChannelOptions) => Array<PresenceMessage>;
+
+    action: PresenceAction;
+    clientId: String;
+    connectionId: String;
+    data: any;
+    encoding: String;
+    id: String;
+    timestamp: number; // Time
+    memberKey: () => String;
+  }
+
+  export declare class ProtocolMessage {
+    action: ProtocolMessageAction;
+    auth: AuthDetails;
+    channel: String;
+    channelSerial: String;
+    connectionDetails: ConnectionDetails;
+    connectionId: String;
+    connectionKey: String;
+    connectionSerial: number;
+    count: number;
+    error: ErrorInfo;
+    flags: String;
+    id: String;
+    messages: Array<Message>;
+    msgSerial: number;
+    presence: Array<PresenceMessage>
+    timestamp: number; // Time
+  }
+
+  export declare class Rest {
+    constructor(options: String | ClientOptions);
+    auth: Auth;
+    channels: Channels;
+    request: (method: string, path: string, parmas?: any, body?: Array<any> | any, headers?: any) => HttpPaginatedResponse;
+    stats: ( start: number, // Time
+           end: number, // Time
+           direction: String,
+           limit: number,
+           unit: String) => PaginatedResult;
+    time: () => number; // time
+  }
+
+  export declare class Realtime {
+    constructor(options: String | ClientOptions);
+
+    auth: Auth;
+    channels: Channels;
+    clientId: String;
+    connection: Connection;
+
+    request: (method: string, path: string, parmas?: any, body?: Array<any> | any, headers?: any) => HttpPaginatedResponse;
+
+    stats: (start: number, // Time
+          end: number, // Time
+          direction: String,
+          limit: number,
+          unit: String) => PaginatedResult;
+
+    close: () => void;
+
+    connect: () => void;
+
+    time: () => number;
+
+  }
+
+  export declare class Connection extends EventEmitter {
+    errorReason: ErrorInfo;
+    id: String;
+    key: String;
+    recoveryKey: String;
+    serial: number;
+    state: ConnectionState;
+
+    close: () => void;
+
+    connect: () => void;
+
+    ping: () => void;
+  }
+
+  export declare class Stats {
+    all: StatsMessageTypes;
+    apiRequests: StatsRequestCount;
+    channels: StatsResourceCount;
+    connections: StatsConnectionTypes;
+    inbound: StatsMessageTraffic;
+    intervalGranularity: StatsIntervalGranularity;
+    intervalId: String;
+    intervalTime: number; // type
+    outbound: StatsMessageTraffic;
+    persisted: StatsMessageTypes;
+    tokenRequests: StatsRequestCount;
+  }
+
+  export declare class ConnectionStateChange {
+  current: ConnectionState;
+  previous: ConnectionState;
+  reason?: ErrorInfo;
+  retryIn: number;
+}
+
+  export declare class PaginatedResult {
+    items: Array<any>;
+
+    first: () => any;
```

Comment:
This method takes a function that has the signature `err ErrorInfo, result PaginatedResult`.  Should this not be defined?